### PR TITLE
ISSUE-7987: The GraphQL topic "Company team mutations" has been created

### LINKED ISSUE #7987: GraphQL: Add company team mutations
# Community engineering issue

Type of Doc:

-  [x] New topic
-  [ ] Topic Update, Devdoc link:
-  [ ] Other

Magento 2 GitHub Issue: https://github.com/magento/magento2/issues/29095

Magento 2 GitHub PR: https://github.com/magento/partners-magento2b2b/pull/227

Magento 2 version:

-  [x] 2.4.2

Notifications of Changes:

-  [ ] Changes to Admin interface?
-  [ ] New software requirements?
-  [x] New REST endpoints or updated GraphQL query syntax?
-  [ ] New command line option?

## Content

Add documentation for the following B2B company team mutations

createCompanyTeam
updateCompanyTeam
deleteCompanyTeam
updateCompanyStructure

### PR DESCRIPTION
**Pull request by andrewbess**
## Purpose of this pull request

This pull request (PR) add a new topic "Company Team mutations" for new 2.4.2 features
This PR hasn't finished yet. It requires the PR about GraphQl query `company` by @keharper 

fixes #7987

### Fixed Issues (if relevant)

1. magento/devdocs#7987: GraphQL: Add company team mutations

### Whatsnew

Added the following queries in support of B2B companies: [createCompanyTeam](https://devdocs.magento.com/guides/v2.4/graphql/mutations/company-team-mutations.html#createcompanyteam-mutation), [updateCompanyTeam](https://devdocs.magento.com/guides/v2.4/graphql/mutations/company-team-mutations.html#updatecompanyteam-mutation), [deleteCompanyTeam](https://devdocs.magento.com/guides/v2.4/graphql/mutations/company-team-mutations.html#deletecompanyteam-mutation), [updateCompanyStructure](https://devdocs.magento.com/guides/v2.4/graphql/mutations/company-team-mutations.html#updatecompanystructure-mutation)

### CODE REVIEW

File: src/_includes/graphql/company-team.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+The `CompanyTeam` object contains details about a company team. It contains the following attributes.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`name` | String | Team name.
+`description` | String | Team description.
+`id` | ID | A string that contains the encoded team id.
```
Comment (rogyar) [inline src/_includes/graphql/company-team.md:7]: It does make sense to set this field as required

```suggestion
`id` | ID! | A string that contains the encoded team id.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+The `CompanyTeamCreateInput` input object defines the company team data.
+
+#### CompanyTeamCreateInput attributes {#CompanyTeamCreateInput}
+
+The `CompanyTeamCreateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`name` | String! | Team Name. This is required attribute.
+`description` | String | Team Description.
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:123]:
```suggestion
`description` | String | Team description.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+The `CompanyTeamUpdateInput` input object defines the company team data.
+
+#### CompanyTeamUpdateInput attributes {#CompanyTeamUpdateInput}
+
+The `CompanyTeamUpdateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id` | ID! | The encoded team id for updating. This is required attribute.
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:204]:
```suggestion
`id` | ID! | The encoded team ID for updating. This is required attribute.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+`name` | String | Team Name.
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:205]:
```suggestion
`name` | String | Team name.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+}
+```
+
+### Input attributes
+
+The `deleteCompanyTeam` mutation requires the following input:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id` | ID! | The encoded team id to delete.
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:266]:
```suggestion
`id` | ID! | The encoded team ID to delete.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+
+The `CompanyStructureUpdateInput` input object defines the company team data.
+
+#### CompanyStructureUpdateInput attributes {#CompanyStructureUpdateInput}
+
+The `CompanyStructureUpdateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`tree_id` | ID! | Company Structure element's hierarchical ID that is being moved to another parent. Required.
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:375]:
```suggestion
`tree_id` | ID! | Company structure element's hierarchical ID that is being moved to another parent. Required.
```

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @andrewbess. Great job! Please, check my comments below. 

Thanks

**Comment by rogyar** [comment]
@andrewbess. Could you also add a "whatsnew" section to the PR description. You may look at an example [here](https://github.com/magento/devdocs/pull/7984). 

Thanks

**Comment by andrewbess** [comment]
Hello @rogyar 
Thank you for your review.
All your suggestions were taken and corrected.
Please check.

Thank you in advance.

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+
+The `CompanyTeamCreateInput` input object defines the company team data.
+
+#### CompanyTeamCreateInput attributes {#CompanyTeamCreateInput}
+
+The `CompanyTeamCreateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`name` | String! | Team Name. This is required attribute.
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:122]: Spotted one more place. Sorry 

```suggestion
`name` | String! | Team name. This is required attribute.
```

**Comment by rogyar** [comment]
Also, @andrewbess, could you check the failing tests, please? 

Thank you!

**Review by keharper — CHANGES_REQUESTED** [review]
@andrewbess Each mutation needs to be in its own topic. 

I've updated the topics for grammar, so be sure to pull in those changes before creating the new files. Also, the `company` object now exists in the `2.4.2-develop` branch, so you can now reference it. It's at `src/_includes/graphql/company.md`

File: src/_data/toc/graphql.yml [hunk 8]
```diff
@@ -185,6 +185,23 @@ pages:
         - label: changeCustomerPassword mutation
           url: /graphql/mutations/change-customer-password.html
 
+        - label: companyTeam mutations
+          url: /graphql/mutations/company-team-mutations.html
+          edition: ee-only
```
Comment (keharper) [inline src/_data/toc/graphql.yml:190]:
```suggestion
          edition: b2b-only
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 9]
```diff
@@ -0,0 +1,382 @@
+---
+group: graphql
+title: Company Team mutations
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:3]: Add the following line in the frontmatter of all B2B topics

b2b_only: true

**Review by keharper — COMMENTED** [review]
I forgot to mention the special tagging for B2B.

**Comment by andrewbess** [comment]
Hello @keharper 
This PR has been fixed according to your suggestions.

File: src/guides/v2.4/graphql/mutations/company-team-mutations.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/company-team-mutations.md:1]: Delete this file. We don't need it any more, now that the mutations are in separate topics.

File: src/guides/v2.4/graphql/mutations/company-team-mutations/create-company-team.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+The `CompanyTeamCreateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`description` | String | An optional description of the team
+`name` | String! | The display name of the team
+`target_id` | ID | The ID of a node within a company's structure. This ID will be the parent of the created team
+
+You can get the `target_id` with the query [`company`]({{page.baseurl}}/graphql/queries/company.html).
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/company-team-mutations/create-company-team.md:113]: Move this sentence to the top of the topic. Also state what happens if you don't specify a target_id.  Is the top node always 0?

File: src/guides/v2.4/graphql/mutations/company-team-mutations/delete-company-team.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
+---
+group: graphql
+title: deleteCompanyTeam mutation
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+b2b_only: true
+---
+
+Use the `deleteCompanyTeam` mutation to delete a company team by ID. You can get the team ID with the qyery [`company`]({{page.baseurl}}/graphql/queries/company.html).
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/company-team-mutations/delete-company-team.md:9]:
```suggestion
Use the `deleteCompanyTeam` mutation to delete a company team by ID. You can get the team ID with the [`company`]({{page.baseurl}}/graphql/queries/company.html) query.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations/update-company-structure.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+### CompanyStructureUpdateInput attributes {#CompanyStructureUpdateInput}
+
+The `CompanyStructureUpdateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`parent_tree_id` | ID! | The ID of a company that will be the new parent
+`tree_id` | ID! | The ID of the company team that is being moved to another parent
+
+You can get the `parent_tree_id` and `tree_id` with the query [`company`]({{page.baseurl}}/graphql/queries/company.html).
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/company-team-mutations/update-company-structure.md:111]: Move this sentence to the top of the topic.

```suggestion
You can get the `parent_tree_id` and `tree_id` with the [`company`]({{page.baseurl}}/graphql/queries/company.html) query.
```

File: src/guides/v2.4/graphql/mutations/company-team-mutations/update-company-team.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+The `CompanyTeamUpdateInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`description` | String | An optional description of the team
+`id` | ID! | The encoded team ID for updating
+`name` | String | The display name of the team
+
+You can get the team ID with the query [`company`]({{page.baseurl}}/graphql/queries/company.html).
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/company-team-mutations/update-company-team.md:77]:
```suggestion
You can get the team ID with the [`company`]({{page.baseurl}}/graphql/queries/company.html) query.
```

Move this sentence to the top

**Review by keharper — CHANGES_REQUESTED** [review]

File: src/guides/v2.4/graphql/mutations/company-team-mutations/create-company-team.md (hunk 11 above)
Comment (andrewbess) [reply src/guides/v2.4/graphql/mutations/company-team-mutations/create-company-team.md]: The `target_id` won't the encoded "0" automatically. There will an encoded value `structure_id` of the main company entity in the value.

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**